benchmark/src/submit_pattern.rs

```diff
@@ -0,0 +1,100 @@
+use logger::prelude::*;
+use std::{thread, time, vec::IntoIter};
+
+/// For constant rate X items per second,
+/// ConstantRate iterator returns 1 item and then sleep for SUBMISSION_TIME_UNIT_US / X us.
+const SUBMISSION_TIME_UNIT_US: u64 = 1_000_000;
+
+/// Utilize IntoIter to implement an iterator that return items at constant speed.
+/// That is, an item is returned to user after waiting some time duration.
+/// Return immediately (e.g. flood all available items) when rate is set to u64::MAX.
+#[derive(Debug, Clone)]
+pub struct ConstantRate<T> {
+    /// Number of items to return per second.
+    constant: u64,
+    /// Time duration to return 1 item.
+    interval_us: u64,
+    /// After return the item, user may perform other operations on the item,
+    /// which consume some time duration X.
+    /// User can set this variable to X so that we can exclude X from the waiting duration.
+    exclude_duration_us: u64,
+    /// Pointer to the items to be consumed.
+    into_iter: IntoIter<T>,
+}
+
+impl<T> ConstantRate<T> {
+    /// Init with constant rate, and the IntoIter of the vector to be consumed.
+    pub fn new(constant: u64, into_iter: IntoIter<T>) -> Self {
+        assert!(constant > 0);
+        let interval_us = SUBMISSION_TIME_UNIT_US / constant;
+        debug!("Return 1 item after {} us", interval_us);
+        ConstantRate {
+            constant,
```

Comment:
I don't see it being used anywhere. Is it needed?

Reply:
It is not explicitly used anywhere, but I think it will useful when user (perhaps not the one who create it) prints the struct and see what is the constant rate.